From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Subject: [PATCH] Bring back __sched_param to bits/sched.h

Upstream removed it because it doesn't use it, but we do: our pthread.h
includes thread-attr.h to bring the pthread_attr_t type to applications,
and thus needs the sched_param structure.

Note: that uses __sched_param and not sched_param because thread-attr.h
is also also brought in from sys/types.h, which should not expose
sched_param if not requested to by applicationg defining _POSIX_SOURCE
and alike.

Signed-off-by: Samuel Thibault <samuel.thibault@ens-lyon.org>

---
 bits/sched.h  | 18 +++++++++++++++++-
 posix/sched.h |  5 ++---
 2 files changed, 19 insertions(+), 4 deletions(-)

diff --git a/bits/sched.h b/bits/sched.h
index 0588f3142e..a75e2c035b 100644
--- a/bits/sched.h
+++ b/bits/sched.h
@@ -17,6 +17,8 @@
    License along with the GNU C Library; if not, see
    <http://www.gnu.org/licenses/>.  */
 
+#ifndef __need_schedparam
+
 #ifndef _BITS_SCHED_H
 #define _BITS_SCHED_H 1
 
@@ -32,7 +34,21 @@
 /* Data structure to describe a process' schedulability.  */
 struct sched_param
 {
-  int sched_priority;
+  int __sched_priority;
 };
 
 #endif /* bits/sched.h */
+
+#endif	/* need schedparam */
+
+#if !defined __defined_schedparam \
+    && (defined __need_schedparam || defined _SCHED_H)
+# define __defined_schedparam	1
+/* Data structure to describe a process' schedulability.  */
+struct __sched_param
+  {
+    int __sched_priority;
+  };
+# undef __need_schedparam
+#endif
+
diff --git a/posix/sched.h b/posix/sched.h
index d5cdac37b0..d139ad27aa 100644
--- a/posix/sched.h
+++ b/posix/sched.h
@@ -43,9 +43,8 @@ typedef __pid_t pid_t;
 #include <bits/sched.h>
 #include <bits/cpu-set.h>
 
-/* Backward compatibility.  */
-#define sched_priority    sched_priority
-#define __sched_priority  sched_priority
+/* Define the real names for the elements of `struct sched_param'.  */
+#define sched_priority    __sched_priority
 
 
 __BEGIN_DECLS
-- 
tg: (854baea9b7..) t2.26/sched_param (depends on: baseline-2.26)
